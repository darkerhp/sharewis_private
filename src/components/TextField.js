import React from 'react';
import { TextInput } from 'react-native';

const { PropTypes } = React;


const TextField = (props) => {
  const {
    style,
    input: { value, onChange },
    meta: { error, touched },
    ...otherProps,
  } = props;

  return (
    <TextInput
      // Let's only change the text color instead of showing error messages
      style={(touched && error) ? [style, { color: 'red' }] : style}
      underlineColorAndroid="transparent"
      onChangeText={text => onChange(text)}
      value={value}
      selectTextOnFocus
      {...otherProps}
    />
  );
};

/* eslint-disable react/forbid-prop-types */
TextField.propTypes = {
  style: PropTypes.object.isRequired,
  meta: PropTypes.object.isRequired,
  input: PropTypes.object.isRequired,
};


export default TextField;
