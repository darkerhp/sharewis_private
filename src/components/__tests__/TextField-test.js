/* eslint-disable no-undef */
/* eslint-disable import/no-extraneous-dependencies */
import 'react-native';
import renderer from 'react-test-renderer';
import React from 'react';
import TextField from '../TextField';

describe('TextField', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <TextField
          input={{
            value: 'text',
            onChange: () => null
          }}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
