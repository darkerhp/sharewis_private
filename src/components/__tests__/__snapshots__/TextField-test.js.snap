exports[`TextField renders correctly 1`] = `
<TextInput
  onChangeText={[Function]}
  selectTextOnFocus={true}
  underlineColorAndroid="transparent"
  value="text" />
`;
